target/linux/ath79/dts/ar9331_glinet_gl-usb150.dts

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "ar9331.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+
+/ {
+	compatible = "glinet,gl-usb150", "qca,ar9331";
+	model = "GL.iNet GL-USB150";
+
+	aliases {
+		serial0 = &uart;
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		label-mac-device = &eth0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wlan {
+			label = "green:wlan";
+			gpios = <&gpio 0 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		led_power: power {
+			label = "green:power";
+			gpios = <&gpio 13 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 11 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+
+		gpio_lan_reset {
+			gpio-export,name = "lan:reset";
+			gpio-export,output = <0>;
+			gpios = <&gpio 7 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	mtd-mac-address = <&art 0x0>;
+
+	gmac-config {
+		device = <&gmac>;
+		switch-phy-addr-swap = <0>;
+		switch-phy-swap = <0>;
+	};
+};
+
+&eth1 {
+	status = "okay";
+
+	compatible = "syscon", "simple-mfd";
+};
+
+&spi {
+	status = "okay";
+
+	num-chipselects = <1>;
```

Comment:
Can somebody enlighten me on num-cs vs. num-chipselects?

Reply:
A quick grep in kernel drivers/spi shows that there's no such a property called num-chipselects.
num-cs is the property to use if a driver wants to specify number of available chipselects in dts.
both spi drivers in ath79 use fixed chipselect count, so num-cs can be omitted.

Reply:
That's interesting. We have about 20 num-chipselects in ath79 and ipq40xx, and actually 171 num-cs in ath79. So, you mean we could remove these for SPI-only devices?

Reply:
num-chipselects is documented as a required property for spi-gpio:
https://www.kernel.org/doc/Documentation/devicetree/bindings/spi/spi-gpio.yaml
but it isn't actually used by the spi-gpio driver. I don't know what's going on with this one.
num-chipselects and num-cs under ath79-spi node can be dropped indeed.

Reply:
Thanks for pointing me towards this. I have removed both num-chipselects and num-cs from spi controller node in ath79 in separate patches:
https://patchwork.ozlabs.org/project/openwrt/patch/20201129160717.1607-1-freifunk@adrianschmutzler.de/
https://patchwork.ozlabs.org/project/openwrt/patch/20201129170348.6077-1-freifunk@adrianschmutzler.de/

I have not touched the definitions actually made for spi-gpio so far.